art_node.go

```diff
@@ -45,19 +45,21 @@ type ArtNode struct {
 	size      uint8
 
 	// Leaf Node Attributes
-	key      []byte
-	keySize  uint64
-	value    interface{}
-	nodeType uint8
+	key         []byte
+	keySize     uint64
+	completeKey []byte
+	value       interface{}
+	nodeType    uint8
 }
 
-func NewLeafNode(key []byte, value interface{}) *ArtNode {
+func NewLeafNode(key, completeKey []byte, value interface{}) *ArtNode {
```

Comment:
Could we just copy the original key instead of exposing a third Parameter to `NewLeafNode` ?  That way we're not putting a requirement on the client to supply the same key twice.